bindinfo/cache.go

```diff
@@ -182,12 +205,86 @@ func (b cache) appendNode(newBindRecord *bindRecord, sparser *parser.Parser) err
 	}
 	stmtNodes, _, err := sparser.Parse(newBindRecord.BindSQL, newBindRecord.Charset, newBindRecord.Collation)
 	if err != nil {
-		return errors.Trace(err)
+		return err
 	}
 	newNode := &bindMeta{
-		bindRecord: newBindRecord,
+		BindRecord: newBindRecord,
 		ast:        stmtNodes[0],
 	}
 	b[hash] = append(b[hash], newNode)
 	return nil
 }
+
+// AddGlobalBind add global bind for SQL.
+func (h *Handle) AddGlobalBind(record *BindRecord) (err error) {
+	ctx := context.TODO() //we need a new ctx to execute a transcation
+	exec, _ := h.ctx.(sqlexec.SQLExecutor)
+	h.Lock()
+	_, err = exec.Execute(ctx, "BEGIN")
+	if err != nil {
+		return
```

Comment:
Before return, we should unlock the lock. The defer statements will not be executed if it is not reached. You can take this as an example:
```go
package main

import "fmt"

func test() {
    i := 100
    defer func() {
        fmt.Printf("before return i=%v\n", i)
    }()
    if i > 10 {
        return
    }

    defer func() {
        fmt.Printf("after return i=%v\n", i)
    }()
}

func main() {
    test()
}
```